### PR TITLE
Add sha1 to lua utils.

### PR DESCRIPTION
**Pull request by basicer**
Raw flag defaulting to false is copied from LuaCrypto's digest function.

### CODE REVIEW

File: src/script/lua_api/l_util.cpp [hunk 1]
```diff
@@ -423,6 +425,31 @@ int ModApiUtil::l_get_version(lua_State *L)
...
+int ModApiUtil::l_sha1(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	std::string data = luaL_checkstring(L, 1);
+	bool hex = !lua_isboolean(L, 2) || !lua_toboolean(L, 2);
+
+	// Compute actual checksum of data
+	std::string data_sha1;
+	{
+		SHA1 data_sha1_calculator;
```
Comment (sfan5) [inline src/script/lua_api/l_util.cpp]: what kind of variable name is this?
just call it `ctx`

Comment (basicer) [reply src/script/lua_api/l_util.cpp]: Good point. I had copied it from somewhere else in minetest where a sha1 was calculated.

File: src/script/lua_api/l_util.cpp [hunk 2]
```diff
@@ -423,6 +425,31 @@ int ModApiUtil::l_get_version(lua_State *L)
...
+		data_sha1_calculator.addBytes(data.c_str(), data.size());
+		unsigned char *data_tmpdigest = data_sha1_calculator.getDigest();
+		data_sha1.assign((char*) data_tmpdigest, 20);
+		free(data_tmpdigest);
+	}
+
+	if (hex) {
+		std::string sha1_hex = hex_encode(data_sha1.c_str());
+		lua_pushlstring(L, sha1_hex.data(), sha1_hex.size());
```
Comment (sfan5) [inline src/script/lua_api/l_util.cpp]: equivalent to `lua_pushtring(L, sha1_hex.c_str());`

Comment (basicer) [reply src/script/lua_api/l_util.cpp]: True, I guess the hex version isn't going to have any nulls.

**Comment by rubenwardy** [comment]
Use case?

**Comment by basicer** [comment]
I'm currently using it with the HTTP api to sign requests.  Seemed odd to have to ship a sha1 implementation in lua with the mod when we already have one inside the engine.

File: src/script/lua_api/l_util.cpp [hunk 3]
```diff
@@ -423,6 +425,31 @@ int ModApiUtil::l_get_version(lua_State *L)
 	return 1;
 }
 
+int ModApiUtil::l_sha1(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	std::string data = luaL_checkstring(L, 1);
```
Comment (SmallJoker) [inline src/script/lua_api/l_util.cpp]: `luaL_checklstring` should be used in case binary data is sent (unwanted NULL termination).

Comment (basicer) [reply src/script/lua_api/l_util.cpp]: Good catch, updated.

**Pull request merged**